src/components/ServiceList.jsx

```diff
@@ -1,22 +1,67 @@
 import { Alert } from "@baltimorecounty/dotgov-components";
 import FilterList from "../components/FilterList";
-import { MostPopularServiceIconStyles } from "../styles";
-import React from "react";
-import ServiceIconLink from "../components/ServiceIconLink";
+import React, { useState } from "react";
+import ServiceIconLink from "./ServiceIconLink";
+import SearchCollapse from "./SearchCollapse";
 import useServices from "../hooks/useServices";
+import {
+  MostPopularServiceIconStyles,
+  LegendText,
+  CollapseContainer
+} from "../styles";
+import Search from "./Search";
 
 const ServiceList = () => {
   const { hasError, serviceItems = [], isLoading } = useServices();
+  const [isChecked, setChecked] = useState(false);
+  const [searchedItems, setSearchedItems] = useState([]);
+  const [searchText, setSearchText] = useState([]);
+  const [isFiltering, setIsFiltering] = useState(false);
 
-  let legendText = {
-    fontStyle: "italic",
-    marginLeft: "5px",
-    fontWeight: "900"
+  const filterItems = (services, searchText) => {
+    return services.filter(item => {
+      const { name, department } = item;
+      return (
+        name.toLowerCase().indexOf(searchText.toLowerCase()) !== -1 ||
+        department.toLowerCase().indexOf(searchText.toLowerCase()) !== -1
+      );
+    });
   };
 
-  let flex = {
-    display: "flex",
-    flexDirection: "row"
+  const filterByPopularity = item => {
+    return item.rank > 0 ? true : false;
+  };
+
+  const checkCondition = (checkedVal, searchText) => {
```

Comment:
i having a little trouble following this.

This runs when the filter changes?

Then its accounting for both filters, and then filtering based on those two things (popular and text input)?

Reply:
Yes. So if popular check box is selected then we use the popular services to filter any text entered. If no text then return only popular services . If Text is entered filter the popular data set against the text entered. If Popular isnt selected then filter the base services against the text. If we add an additional filter my head might explode trying to figure out what we are filtering so we may need to come up with a better way to handle this? I think it might be better to send the actual list we want filtered up instead of trying to figure out which one we are currently filtering.